presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -154,6 +154,7 @@
 import static com.facebook.presto.hive.metastore.StorageFormat.VIEW_STORAGE_FORMAT;
 import static com.facebook.presto.hive.metastore.StorageFormat.fromHiveStorageFormat;
 import static com.facebook.presto.hive.util.ConfigurationUtils.toJobConf;
+import static com.facebook.presto.hive.util.StatisticsUtils.NUM_PARTITIONS;
```

Comment:
> Store partitions count in table properties

Same general comments as in `Store table level statistics for partitioned tables`:
- can affect concurrency
- can go out of sync if there are other writers to metastore

And also:
- https://cwiki.apache.org/confluence/display/Hive/StatsDev looks as if this was not supposed to be set

Reply:
> can affect concurrency

It does not affect concurrency right now. Actually there is no locking in SemiTransactionalHiveMetastore that affects multiple transactions.

But the problem is that, yeah

> can go out of sync if there are other writers to metastore

I'm going to change `updateTableParameter` methods to be `compareAndSet` to do not loose any updates.